wall_e/src/resources/cogs/ban.py

```diff
@@ -101,13 +101,15 @@ async def intercept(self, guild: discord.Guild, member: Union[discord.User, disc
         # need to read the audit log to grab mod, date, and reason
         logger.info(f"[Ban intercept()] guild ban detected and intercepted for user='{member}'")
         try:
-            audit_ban = await self.bot.guilds[0].audit_logs(action=discord.AuditLogAction.ban).flatten()
+            date = datetime.datetime.now() - datetime.timedelta(minutes=1)
+            audit_ban = await guild.audit_logs(action=discord.AuditLogAction.ban, after=date,
+                                               oldest_first=False).flatten()
         except Exception as e:
             logger.info(f'error while fetching ban data: {e}')
             await self.mod_channel.send(f"Encountered following error while intercepting a ban: {e}\n" +
                                         "**Most likely need view audit log perms.**")
             return
-        audit_ban = audit_ban[0]
+        audit_ban = next(user for user in audit_ban if user.target.id == member.id)
```

Comment:
`for user in audit_ban` makes no sense 🤔 
it should either be `for audit_ban in audit_bans` or `for user in banned_user`